### PR TITLE
update some of my parts

### PR DESCRIPTION
**Pull request by l3naz**
# Description

What do your changes do?

# Area of Effect

What section of the code base does your changes effect?

### CODE REVIEW

**Comment by l3naz** [comment]
Just update some of my part

File: src/core/opcode.c [hunk 1]
```diff
@@ -38,17 +38,32 @@ void chip8_opcode_2NNN(chip8_cpu *cpu)
...
+                cpu->pc += 2;
+            }
 }
 
 void chip8_opcode_5XY0(chip8_cpu *cpu)
 {
-	/* TODO */
+	uint8_t x = (cpu->opcode & 0x0F00) >> 8;
+	uint8_t y = (cpu->opcode & 0x00F0) >> 4;
+	if (cpu->v[x] == cpu->cpu.v[y])
```
Comment (awkless) [inline src/core/opcode.c]: Syntax error.

You meant to type:

```
if (cpu->v[x] == cpu->v[y])
```

**Review by awkless — CHANGES_REQUESTED** [review]
You have a syntax error at line 63. Plus, you are missing your implementation of opcode 8XY5.

File: src/core/opcode.c [hunk 2]
```diff
@@ -38,17 +38,32 @@ void chip8_opcode_2NNN(chip8_cpu *cpu)
...
+                cpu->pc += 2;
+            }
 }
 
 void chip8_opcode_5XY0(chip8_cpu *cpu)
 {
-	/* TODO */
+	uint8_t x = (cpu->opcode & 0x0F00) >> 8;
+	uint8_t y = (cpu->opcode & 0x00F0) >> 4;
+	if (cpu->v[x] == cpu->v[y]))
```
Comment (awkless) [inline src/core/opcode.c]: You have a left over parenthesis.

**Review by awkless — CHANGES_REQUESTED** [review]
You have a syntax error on line 63, you have a left over parenthesis.

File: src/core/opcode.c [hunk 3]
```diff
@@ -69,32 +84,56 @@ void chip8_opcode_7XNN(chip8_cpu *cpu)
...
+	uint8_t y = (cpu->opcode & 0x00F0) >> 4;
+	cpu->v[x] = cpu->v[x] ^ cpu->v[y];
 }
 
 void chip8_opcode_8XY4(chip8_cpu *cpu)
 {
-	/* TODO */
+	uint8_t x = (cpu->opcode & 0x0F00) >> 8;
+	uint8_t y = (cpu->opcode & 0x00F0) >> 4;
+	tmp = cpu->v[x] + cpu->v[y];
```
Comment (awkless) [inline src/core/opcode.c:118]: Variable tmp is undeclared. You probably meant:

```
uint16_t tmp = cpu->v[x] + cpu->v[y];
```

**Review by awkless — CHANGES_REQUESTED** [review]

File: src/core/opcode.c [hunk 4]
```diff
@@ -69,32 +84,58 @@ void chip8_opcode_7XNN(chip8_cpu *cpu)
...
+            cpu->v[0x0f] = true;
+        }
+
+        cpu->v[x] = tmp;
 }
 
 void chip8_opcode_8XY5(chip8_cpu *cpu)
 {
-	/* TODO */
+	uint16_t tmp = 0;
```
Comment (awkless) [inline src/core/opcode.c]: The `tmp` variable is defined but never used in the scope of this function. I recommend removing it.

File: src/core/opcode.c [hunk 5]
```diff
@@ -69,32 +84,58 @@ void chip8_opcode_7XNN(chip8_cpu *cpu)
...
+	uint8_t x = (cpu->opcode & 0x0F00) >> 8;
+	uint8_t y = (cpu->opcode & 0x00F0) >> 4;
+	cpu->v[0x0f] = false;
```
Comment (awkless) [inline src/core/opcode.c]: According to the CHIP-8 specification the VF register must be set to true if no borrow occurs.

File: src/core/opcode.c [hunk 6]
```diff
@@ -69,32 +84,58 @@ void chip8_opcode_7XNN(chip8_cpu *cpu)
...
+        if (cpu->v[x] > cpu->v[y])
+        {
+            cpu->v[0x0f] = true;
```
Comment (awkless) [inline src/core/opcode.c]: According to the CHIP-8 specification the VF register must be set to false in case of borrow, i.e., underflow.

**Review by awkless — CHANGES_REQUESTED** [review]
Your implementation of opcode 8XY5 does not properly set the VF register. You need to set the VF register to true if __no__ underflow (borrow) occurs. I recommend doing the following:

```
if (cpu->v[x] > cpu->v[y])
        cpu->v[0x0F] = false; // Borrow occurred, set to false...
else
        cpu->v[0x0F] = true; // No borrow occurred, set to true...
```
Plus, you left the `tmp` variable in your opcode 8XY5 implementation unused. I recommend that you remove it.

File: src/core/opcode.c [hunk 7]
```diff
@@ -69,32 +84,59 @@ void chip8_opcode_7XNN(chip8_cpu *cpu)
...
+	uint8_t x = (cpu->opcode & 0x0F00) >> 8;
+	uint8_t y = (cpu->opcode & 0x00F0) >> 4;	
+	if (cpu->v[x] > cpu->v[y])
+	{
+        	cpu->v[0x0F] = false; 
+	}
+	else
+	{
+        	cpu->v[0x0F] = true;
+	}
```
Comment (awkless) [inline src/core/opcode.c:139]: What happened to the actual subtraction operation?

**Review by awkless — CHANGES_REQUESTED** [review]
You forgot to actually perform the subtraction for opcode 8XY5:

```
cpu->v[x] = cpu->v[x] - cpu->v[y];
```

**Review by awkless — APPROVED** [review]

**Pull request merged**